rust/sdk/src/client.rs

```diff
@@ -282,86 +263,33 @@ impl EventService for ActyxClient {
         })?)
     }
 
-    /// Publishes a set of new events.
-    ///
-    /// If an authorization error (code 401) is returned, it will try to re-authenticate.
-    /// If the service is unavailable (code 503), this method will retry to perform the
-    /// request up to 10 times with exponentially increasing delay - currently,
-    /// this behavior is only available if the `with-tokio` feature is enabled.
-    async fn publish(&self, request: PublishRequest) -> anyhow::Result<PublishResponse> {
-        let body = serde_json::to_value(&request).context(|| format!("serializing {:?}", &request))?;
-        let response = self
-            .do_request(|c| c.post(self.events_url("publish")).json(&body))
-            .await?;
-        let bytes = response
-            .bytes()
-            .await
-            .context(|| format!("getting body for GET {}", self.events_url("publish")))?;
-        Ok(serde_json::from_slice(bytes.as_ref()).context(|| {
-            format!(
-                "deserializing publish response from {:?} received from GET {}",
-                bytes,
-                self.events_url("publish")
-            )
-        })?)
+    /// Returns a builder for publishing events.
```

Comment:
Please add an example showing that the Builder also is a Future, i.e. can be `.await`ed. I’d use a single event to demo it because that will be the most frequent usage.

Reply:
I was going to write a full one, but since we didn't merge #557 yet, I didn't to avoid duplicate work 😅